.github/workflows/typecheck.yml

```diff
@@ -29,5 +29,8 @@ jobs:
       - name: Install web dependencies
         if: steps.cache.outputs.cache-hit != 'true'
         run: pnpm install --frozen-lockfile
+      - name: Generate config files
+        if: steps.cache.outputs.cache-hit == 'true'
```

Comment:
```suggestion
        id: generated-files
        if: steps.generated-files.outputs.cache-hit != 'true'
```

Reply:
Not 100% on this but looking at the docs [here](https://github.com/actions/cache#using-a-combination-of-restore-and-save-actions) it seemed reasonable

Reply:
If this works we should update the other workflows that generate these files to share the cache id

Reply:
The cache will be shared by the workflows but the ID will not be, so it needs a unique identifier (like hashing the lockfiles) otherwise we can't guarantee it's the latest cache as they don't update if there is a cache hit. There are some ways around this though such as adding the run id which will always be different then using fallback keys for the current branch and master.